client/keys.go

```diff
@@ -25,6 +29,12 @@ type Key struct {
 	rw      io.ReadWriter
 	handle  tpmutil.Handle
 	pubArea tpm2.Public
+	// >>> Start Direct Implementaion <<<
+	transportTPM  transport.TPM
+	pubAreaDirect tpmt.Public
+	nameDirect    *tpm2b.Name
```

Comment:
Rather than duplicating these fields, could we have funcs that do the conversions between legacy and direct?

There's no getting around the different session, but I don't think we should duplicate the data for the TPM structs or the TPM interface (transport.TPM). That could lead to data inconsistency issues.

Reply:
After discussing with Joe yesterday, we concluded that back compatibility would not be possible thus why I did not choose to have conversion between legacy and direct. The reason why it is duplicated so that it will be easier to deprecate old legacy code later on.

Reply:
Chatted offline and decided to have the immutable fields (name/pubArea) duplicated but stored in `finish()`. Then we will expose the interface to the TPM with a private `Key.transport()` method so that we can migrate over to the new `transport.TPM` type